src/summersplash2022/Content.tsx

```diff
@@ -0,0 +1,413 @@
+import React, { useRef, useState } from 'react';
+import NewVariants from './img/newVariants';
+import Map from './img/map';
+import style from './index.module.css';
+import HandBooKPages from './utils/handBookPages';
+import { JobOrInternship } from './utils/utils';
+import TimeLine from './img/tidslinje';
+import ApplyAsDesigner from './img/søkSomDesigner';
+import ApplyAsDesignerBiggerBlob from './img/søkSomDesignerBigBlob';
+import ApplyAsDeveloper from './img/søkSomUtviklerBlob';
+import ApplyAsDeveloperBiggerBlob from './img/søkeSomUtviklerBigBlob';
+import ReadMoreArrow from './img/lesMerPil';
+
+const Content = ({ selected }: JobOrInternship) => {
+  const blobSection1 = require('./img/section1Blob.png');
+  const Marius = require('./img/Marius.png');
+  const summerStudent1 = require('./img/Vilde summerstudent.png');
+  const summerStudent2 = require('./img/Olepetter summerstudent.png');
+  const summerStudent3 = require('./img/Nora summerstudent.png');
+
+  const [whichButtonSelected, setWhichButtonSelected] = useState('');
+
+  const forside = useRef<HTMLImageElement>(null);
+  const sommerjobbHandling = useRef<HTMLImageElement>(null);
+  const sommerjobbereMening = useRef<HTMLImageElement>(null);
+  const hvorforVariant = useRef<HTMLImageElement>(null);
+  const etterSoknad = useRef<HTMLImageElement>(null);
+  const sokSommerjobb = useRef<HTMLImageElement>(null);
+
+  const goToRef = (ref: React.RefObject<HTMLImageElement>) => {
+    ref?.current?.scrollIntoView({ behavior: 'smooth' });
+  };
+
+  // Undefined for now. Need to be changed to 'internship' when index sends a prop.
+  if (selected === 'internship') {
+    return (
+      <>
+        <div className={style.scrollContainer}>
+          <section className={style.section1} ref={forside}>
+            <span className={style.searchNewVariants}>
+              <NewVariants />
+            </span>
+            <h4 className={style.h4Section1}>Søk innen 2. Oktober</h4>
+
+            <br />
+            {/* Navigation bar on the right side */}
+            <div className={style.navigationSlider}>
+              <button
+                className={style.navigationButton6}
+                onClick={() => goToRef(forside)}
```

Comment:
Her kunne det vært mye enklere med å rett og slett bruke vanlige `<a>` lenker og anchor tags. Eksempel

```tsx
<a href="#forside" className={style.navigationButton6}>Forside</a>

// og der `ref={forside}` er lengre opp:

<section className={style.section1} id="forside">
```

Da benytter vi oss av innebygget navigasjon i nettleseren som setter hvor den skal dukke opp. Det blir i praksis som variant.no/#forside, og da scroller den ned til blokken hvor forside finnes. Se eksempler og dokumentasjon på det her: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#linking_to_an_element_on_the_same_page

Da misser du smooth scrolling som det er brukt her, men det kan vi enkelt fikse med litt CSS. Med å sette dette på container som scroller, som f.eks:

```css
body {
  scroll-behavior: smooth;
}
```

Se eksempel og docs her: https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-behavior


Effekten av det er mye mindre ting å manuelt kode, vi benytter oss av innebygget funksjonalitet i nettleseren (som som regel er en fordel for det betyr at du får mye gratis som universell utforming, navigasjon via høyre-museklikk -> åpne i ny tab, osv). Mindre å kode, men samtidig bedre resultat. Win-win! 👯